packages/gatsby-remark-autolink-headers/src/__tests__/index.js

```diff
@@ -182,4 +183,42 @@ describe(`gatsby-remark-autolink-headers`, () => {
       expect(node.data.id).toEqual(expect.stringMatching(/^heading/))
     })
   })
+
+  it(`extracts custom id`, () => {
+    const markdownAST = remark.parse(`
+# Heading One {#custom_h1}
+
+## Heading Two {#custom-heading-two}
+
+# With *Bold* {#custom-withbold}
+
+# Invalid {#this_is_italic}
+    `)
+    const enableCustomId = true
+
+    const transformed = plugin({ markdownAST }, { enableCustomId })
+
+    const headers = []
+    visit(transformed, `heading`, node => {
+      headers.push({ text: toString(node), id: node.data.id })
+    })
+    expect(headers).toEqual([
+      {
+        id: `custom_h1`,
+        text: `Heading One`,
+      },
+      {
+        id: `custom-heading-two`,
+        text: `Heading Two`,
+      },
+      {
+        id: `custom-withbold`,
+        text: `With Bold`,
+      },
+      {
+        id: `invalid-thisisitalic`,
```

Comment:
I'm no markdown expert but why wouldn't I want to get the following?
```json
{
  "id: "this_is_italic",
  "text": "Invalid",
}
```

Reply:
Because `# Invalid {#this_is_italic}` is parsed as `<h1>Invalid {#this<i>is</i>italic}</h1>` (Three child nodes under <h1>) before this plugin.
The current implementation extracts `{#id}` only from the last child of header nodes.

Reply:
Oh because we convert markdown to html. Maybe that's a mistake, maybe we should use the raw value instead to create the idea. Nonetheless, we can iterate on this in a follow PR when people are complaining about it.

Thanks for the explanation! 💯